pkg/server/plugin/upstreamauthority/vault/vault_client.go

```diff
@@ -350,10 +350,19 @@ func (c *Client) LookupSelf(token string) (*vapi.Secret, error) {
 func (c *Client) SignIntermediate(ttl string, csr *x509.CertificateRequest) (*SignCSRResponse, error) {
 	csrPEM := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE REQUEST", Bytes: csr.Raw})
 
+	var uris []string
+	for _, uri := range csr.URIs {
+		uris = append(uris, uri.String())
+	}
+	if len(uris) == 0 {
+		return nil, status.Errorf(codes.Internal, "CSR must have least one URIs")
```

Comment:
Since the error is due to an invalid input argument and not an internal failure I think that we should use codes.InvalidArgument.

```suggestion
		return nil, status.Errorf(codes.InvalidArgument, "CSR must have at least one URI")
```

Reply:
fixed via [f9b6103](https://github.com/spiffe/spire/pull/3971/commits/f9b610336ff31073fc7ba84b1b68d120fc8ff1ed)